runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -172,15 +175,27 @@ public void commit() {
      * Otherwise this throws a TransactionAbortedException.
      *
      * @param timestamp Timestamp to commit the transaction on.
+     * @param numRetries number of times to retry the transaction on internal errors.
      */
-    public void commit(final Timestamp timestamp) {
-        this.timestamp = timestamp;
-        try {
-            txBegin();
-            operations.forEach(Runnable::run);
-        } finally {
-            txEnd();
-            operations.clear();
+    public void commit(final Timestamp timestamp, int numRetries) {
```

Comment:
if you make `DEFAULT_NUM_TX_RETRIES` non-static, then you don't need to pass  `numRetries` as a parameter.
And you can check that DEFAULT_NUM_TX_RETRIES is correct in TxBuilder's constructor. 
Which makes your code more resilient and secure

Reply:
done. thanks